### PR TITLE
[flutter_tools] remove globals from base/android

### PR DESCRIPTION
**Pull request by jonahwilliams**
## Description

Remove unnecessary use of globals from base tests, and an android test. This changes the test to avoid bouncing through the global getters, which can lead to incorrectly cached zone values. Switches the memory filesystem implementation to the test implementation

### CODE REVIEW

File: packages/flutter_tools/test/general.shard/android/android_sdk_test.dart [hunk 1]
```diff
@@ -4,10 +4,13 @@
...
 import 'package:file/memory.dart';
 import 'package:flutter_tools/src/android/android_sdk.dart';
+import 'package:flutter_tools/src/base/config.dart';
 import 'package:flutter_tools/src/base/file_system.dart';
 import 'package:flutter_tools/src/base/io.dart' show ProcessResult;
+import 'package:flutter_tools/src/base/logger.dart';
 import 'package:flutter_tools/src/base/platform.dart';
-import 'package:flutter_tools/src/globals.dart' as globals;
+import 'package:meta/meta.dart';
+// import 'package:flutter_tools/src/globals.dart' as globals;
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/android/android_sdk_test.dart:13]: delete?

Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/android/android_sdk_test.dart]: Fixed!

File: packages/flutter_tools/test/general.shard/base/os_utils_test.dart [hunk 2]
```diff
@@ -3,42 +3,46 @@
...
+        processManager: const LocalProcessManager(),
+      );
+      final File file = fileSystem.file(fileSystem.path.join(tempDir.path, 'foo.script'));
       file.writeAsStringSync('hello world');
-      globals.os.makeExecutable(file);
+      operatingSystemUtils.makeExecutable(file);
 
       // Skip this test on windows.
-      if (!globals.platform.isWindows) {
+      if (!platform.isWindows) {
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/base/os_utils_test.dart:41]: I know this is copying existing functionality, but should we skip earlier if we're on windows?

File: packages/flutter_tools/test/general.shard/base/terminal_test.dart [hunk 3]
```diff
@@ -42,7 +41,7 @@ void main() {
 
     setUp(() {
       terminal = AnsiTerminal(
-        stdio: globals.stdio, // Danger, using real stdio.
+        stdio: Stdio(), // Danger, using real stdio.
```
Comment (christopherfujino) [inline packages/flutter_tools/test/general.shard/base/terminal_test.dart:44]: doesn't have to be here, but from looking at this code it looks like we could replace this with a mock.

**Review by christopherfujino — APPROVED** [review]
LGTM

File: packages/flutter_tools/test/general.shard/base/os_utils_test.dart (hunk 2 above)
Comment (jonahwilliams) [reply packages/flutter_tools/test/general.shard/base/os_utils_test.dart:41]: I think this is sort of - badly testing that chmod doesn't throw.

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/base/os_utils_test.dart:41]: Oh, I missed that this was using the real file system. `chmod` is a no-op on Windows in any case...`  void chmod(FileSystemEntity entity, String mode) {}`

Comment (christopherfujino) [reply packages/flutter_tools/test/general.shard/base/os_utils_test.dart:41]: makeExecutable too:

`  void makeExecutable(File file) {}`

**Pull request merged**